client/my-sites/stats/feedback/modal/index.tsx

```diff
@@ -19,6 +22,23 @@ interface ModalProps {
 
 const NOTICE_KEY_FOR_FEEDBACK_SUBMISSION = 'able_to_submit_user_feedback';
 
+const FEEDBACK_SHOULD_SHOW_PANEL_API_KEY = NOTICES_KEY_SHOW_FLOATING_USER_FEEDBACK_PANEL;
+const FEEDBACK_SHOULD_SHOW_PANEL_API_HIBERNATION_DELAY = 60; // 30 seconds for now
+// Examples values:
+// 30 minutes = 60 * 30;
+// 30 days = 3600 * 24 * 30;
+
+function useFeedbackHibernationMutation( siteId: number ) {
```

Comment:
Is the function really necessary? It only wraps up a single call and returns it as a different name. In my opinion, the name of the variable is already clear enough `updateFeedbackHibernationPeriod` and we don't have to use another function.

If we are all using react-query in the same way, we don't have to dig into the wrapper, which would make the review easier and bundle smaller imho.

Reply:
I'd say this is definitely a matter of opinion. My intention was to keep it clear without additional refactoring but rather than remove this, I'd prefer to move the other Notices hooks into this. That allows us to remove unnecessary noise from the main component. For example, the refetch code. To my mind, we'd call one hook for our use of the Notices API and one for the submission API. An example of my personal preference would be this:

https://github.com/Automattic/wp-calypso/blob/2e76ffcea78ba7ea6389feaf7cf907404d541f5a/client/my-sites/stats/feedback/index.tsx#L27

In that example, details of the hooks are separated out and I think it makes their usage in the component much cleaner.

Reply:
I pushed a temp branch so it would be easier to show what I'm meaning here.

https://github.com/Automattic/wp-calypso/pull/94463

The specific changes I would make would be the following commit:

https://github.com/Automattic/wp-calypso/pull/94463/commits/ff720bf825c7072e405eeca737b7714ce537007d

Let me know what you think. I feel pretty strongly that this improves readability but if the team disagrees, I'm happy to put things inline in the main component.

Reply:
I've moved the code back inline. I'd still be curious to hear what you think of the approach linked above though. Thanks! 👍

Reply:
I actually like the two examples you shared. My comment was only to the specific case in the PR, but if the function has other things to do rather than wrapping up a single call, then I'd be happy.

@dognose24 @annacmc  what do you think?

Reply:
Sorry for not participating in the discussion earlier. I prefer the inline approach because, in my opinion, too many functions would also be noises if they were not actually shared. cc @kangzj @a8ck3n